### DISCUSSION TITLE
File tracing

### DISCUSSION THREAD

**Discussion by egoist** [egoist/tsup#506] [Ideas]
esbuild can only bundle `import/require`, so it doesn't work for `update-notifier` package which includes code like this:

```js
spawn(process.execPath, [path.join(__dirname, 'check.js')])
```

`check.js` is not copied to out dir, tsup could introduce a `--trace-files` flag to support this 🧐

**Comment by OmgImAlexis** [comment]
In a similar sense there are files that use imports like so for native files which I would love to see resolved at build time. This would allow tsup to then ship the `.node` file. Right now it skips the `.node` file resulting in the bundled file not working.

```js
return require('./uws_' + process.platform + '_' + process.arch + '_' + process.versions.modules + '.node');
```
